jax/experimental/sparse/ops.py

```diff
@@ -979,6 +979,14 @@ def _bcoo_transpose_batch_rule(batched_args, batch_dims, *, permutation, shape):
 
 bcoo_dot_general_p = core.Primitive('bcoo_dot_general')
 
+def _dot_general_validated_shape(lhs_shape, rhs_shape, dimension_numbers):
```

Comment:
From my experience reading jax code. Since the variable names are not descriptive, maybe add type annotations here?

(The variable names here are descriptive, but was saying in general type annotations help a lot)

Reply:
Done, thanks!